### PR TITLE
Staking sdk

### PR DESCRIPTION
**Pull request by chenshoo**
## Pull Request Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes (link to the issue here)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Chore / Documentation
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Locally tested against Fireblocks API

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added corresponding labels to the PR

### CODE REVIEW

**Review by orgranit — APPROVED** [review]

**Review by YoavBZ — CHANGES_REQUESTED** [review]
After discussing this internally - new methods should be added to the `FireblocksSDK` class and not in a new one. We should keep the current approach of not introducing new namespaces (NCW is exceptional).
Please align this accordingly

**Comment by orgranit** [comment]
> After discussing this internally - new methods should be added to the `FireblocksSDK` class and not in a new one. We should keep the current approach of not introducing new namespaces (NCW is exceptional). Please align this accordingly

@YoavBZ 
Can you share the reasoning behind it? 
fireblocks-sdk.ts is already 2000 lines long, and nameSpaces help us prevent name collisions. 
For example I prefer to keep our methods name simple as `sdk.staking.getChains` or `sdk.staking.approveProviderTermsOfService` if we would be force to flatten all under the general sdk, it would have to be called `sdk.getStakingChains` or `sdk.approveStakingProviderTermsOfService`.

File: src/fireblocks-sdk.ts [hunk 1]
```diff
@@ -188,14 +199,60 @@ export class FireblocksSDK {
...
+    /**
+     * Get all staking positions, optionally filtered by chain
+     */
+    public async getStakingPositions(chainDescriptor?: StakingChain): Promise<StakingPosition[]> {
+        return await this.stakingApiClient.getPositions(chainDescriptor);
+    }
+    /**
+     * Get a staking position by id
+     */
+    public async getStakingPosition(positionId?: string): Promise<StakingPosition[]> {
```
Comment (chenshoo) [inline src/fireblocks-sdk.ts:241]: why is it optional?

File: src/fireblocks-sdk.ts [hunk 2]
```diff
@@ -188,14 +199,60 @@ export class FireblocksSDK {
...
+    /**
+     * Get staking positions summary
+     */
+    public async getStakingPositionsSummary(byVault?: boolean): Promise<DelegationSummaryDto | DelegationSummaryDtoByVault> {
+        return await this.stakingApiClient.getPositionsSummary(byVault);
+    }
+    /**
+     * Execute staking action on a chain
+     */
+    public async executeStakingAction(actionId: StakingAction, chainDescriptor: StakingChain, body: any): Promise<ExecuteActionResponse> {
```
Comment (chenshoo) [inline src/fireblocks-sdk.ts:229]: maybe create type for body that's stakeRequestDTO | unstakeRequestDTO | withdrawRequestDTO

Comment (yahavamar) [reply src/fireblocks-sdk.ts]: Done

File: src/fireblocks-sdk.ts (hunk 1 above)
Comment (yahavamar) [reply src/fireblocks-sdk.ts]: Fixed

**Review by aviba — APPROVED** [review]

**Review by aviba — APPROVED** [review]

**Review by aviba — APPROVED** [review]

**Pull request merged**